### PR TITLE
 Add categories api 

### LINKED ISSUE #8: Add api for fetching categories
API: "/categories"
Get all categories with their associated tags sorted by name.

### PR DESCRIPTION
**Pull request by pratap0007**
This PR has categories api which returns all categories with their associated tags sorted by name.

- add api structure for categories api
- replace log by sugared logger
- add gorm connection
- add database migration for category api
- add category api implementation and gorm test

closes #8

### CODE REVIEW

File: api/test/category_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	categoryservice "github.com/tektoncd/hub/api/pkg/service"
+)
+
+var (
+	categorySvc category.Service
+)
+
+// test for caetgory endpoint using gorm connection and load fixtures
+func TestGetCategories(t *testing.T) {
+	checker := goahttpcheck.New()
```
Comment (SM43) [inline api/test/category_test.go:22]: We will have to call `LoadFixture` before executing every test.

File: api/test/category_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			}
+			defer r.Body.Close()
+			var jsonMap []map[string]interface{}
+			err = json.Unmarshal([]byte(b), &jsonMap)
+			if err != nil {
+				panic(err)
+			}
+			assert.Equal(t, len(jsonMap), 3)
+			assert.Equal(t, jsonMap[0]["name"], "abc")
+		})
```
Comment (SM43) [inline api/test/category_test.go:41]: can you add `assert` to check whether tags are preloaded?

File: api/pkg/service/category_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	return nil
+}
+
+func Test_All(t *testing.T) {
+	categorySvc = NewCategory(testConfig)
+	LoadFixture(db, "../../fixtures")
+	all, err := categorySvc.All(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, len(all), 3)
+	assert.Equal(t, all[0].Name, "abc") // categories are sorted by name
```
Comment (SM43) [inline api/pkg/service/category_test.go:39]: Can you please add a `assert` to check a tags are preloaded?

File: api/fixtures/categories.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+- id: 1
+  name: pqr
```
Comment (PuneetPunamiya) [inline api/fixtures/categories.yaml:2]: Can we have some better names

File: api/fixtures/tags.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+- id: 1
+  name: ab
```
Comment (PuneetPunamiya) [inline api/fixtures/tags.yaml:2]: Can we have some better names?

File: api/pkg/db/model/migration.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+)
+
+// Migrate create tables and populates master tables
+func Migrate(api *app.ApiConfig) error {
+
+	logger := api.Logger()
+
+	// NOTE: If writing a migration for a new table then add the same in InitSchema
+	migration := gormigrate.New(api.DB(), gormigrate.DefaultOptions, []*gormigrate.Migration{
+		// NOTE: Add Migration Here
```
Comment (PuneetPunamiya) [inline api/pkg/db/model/migration.go:16]: Is this comment necessary ?

Comment (SM43) [reply api/pkg/db/model/migration.go:16]: Lets keep the comment.

File: api/gen/http/openapi.yaml [hunk 7]
```diff
@@ -13,28 +13,137 @@ produces:
...
+          name: cli
+        - id: 1
+          name: cli
+    example:
+      id: 1
+      name: Image-build
+      tags:
+      - id: 1
+        name: cli
+      - id: 1
```
Comment (sthaha) [inline api/gen/http/openapi.yaml:144]: the id repeats, is that on purpose?

File: api/design/type.go [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	Attribute("id", UInt, "unique id of category", func() {
+		Example("id", 1)
+	})
+	Attribute("name", String, "name of category", func() {
+		Example("name", "Image-build")
+	})
+	Attribute("tags", ArrayOf(ResourceTags), "list of tag associated with category")
+	Required("id", "name", "tags")
+})
+var ResourceTags = Type("Tag", func() {
```
Comment (SM43) [inline api/design/type.go:17]: ResourceTag? Singular?

File: api/gen/http/openapi.yaml (hunk 7 above)
Comment (pratap0007) [reply api/gen/http/openapi.yaml:144]: no,it repeats for Attribute having type ArrayOf()

File: api/pkg/db/model/model.go [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+		Versions  []ResourceVersion
+		Tags      []*Tag `gorm:"many2many:resource_tags;"`
+	}
+
+	// ResourceVersion represents diffrent versions of a Resource
+	ResourceVersion struct {
+		gorm.Model
+		Version     string
+		Description string
+		URL         string
```
Comment (SM43) [inline api/pkg/db/model/model.go:54]: Can you add `DisplayName`?

File: api/pkg/app/app.go [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+	"github.com/jinzhu/gorm"
+	"github.com/joho/godotenv"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	// Blank for package side effect: loads postgres drivers
+	_ "github.com/lib/pq"
+)
+
+type Base interface {
```
Comment (sthaha) [inline api/pkg/app/app.go:17]: Could we merge the Base and the Config until we have a good reason to split the interfaces (like the experimental/hub/api) ?

**Pull request merged**